### PR TITLE
Rb 123 fix random bullet out of bound

### PR DESCRIPTION
**Pull request by kw4ntiq**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] Norm of the code has been respected

* **In what state is your pull request?**
- [ ] Ready to merge / Waiting a reviwer to see my work.
- [x] Work In Progress (WIP) / My work is not finish but i want daily reviews. (Draft)
- [ ] CI Review / I want to see what the CI thinks of my work.
- [ ] Need feedback / Just to have feedback on what i produced. (May not be merged)

* **What kind of change does this PR introduce?** (You can choose multiple)
- [x] Bug fix
- [ ] Feature request
- [ ] New / Updated documentation
- [ ] Testing CI ( Make the pull request in draft mode)

* **What is the current behavior?** (link an issue based on the kind of change this pr introduce)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### Summary by CodeRabbit

**Release Notes**

- **New Feature**: Enhanced bullet shooting mechanism in the game. Players can now only shoot bullets if they have positive health and if the space key is pressed. This adds a layer of strategy and realism to the gameplay.
- **Refactor**: Improved the handling of entities after death, making the game more stable and efficient.
- **New Feature**: Updated key bindings for playing music and sound effects, providing a more intuitive user experience.
- **Bug Fix**: Fixed a bug where sprites were drawn without checking for their existence, preventing potential crashes.
- **Refactor**: Optimized the handling of health updates and position management for entities, improving game performance.
- **Refactor**: Updated the parallax system for a smoother visual experience.
- **Style**: Introduced meaningful variable and constant names for better code readability and maintenance.

### CODE REVIEW

File: src/Client/Systems/Events/EventsSystems.cpp [hunk 1]
```diff
@@ -70,24 +70,30 @@ namespace Systems {
...
-                struct Types::Missiles missile = {
-                    .type = CLASSIC,
-                };
-                createMissile(arrPosition[id], missile);
+        if (Raylib::isKeyDown(Raylib::KeyboardKey::KB_SPACE) == false
+            || clock_.elapsedMillisecondsSince(clockId) < waitTimeBullet) {
+            return;
+        }
+
+        clock_.restart(clockId);
```
Comment (Saverio976) [inline src/Client/Systems/Events/EventsSystems.cpp]: clock_.decreaseMilliseconds(clockId, waitTimeBullet)

Comment (guillaumeAbel) [reply src/Client/Systems/Events/EventsSystems.cpp]: this function dont even exist here, its in my pr

**Review by Saverio976 — APPROVED** [review]

**Review by Saverio976 — APPROVED** [review]

**Review by romainpanno — APPROVED** [review]

File: src/ECS/Systems/Systems.cpp [hunk 2]
```diff
@@ -293,25 +294,25 @@ namespace Systems {
...
-        for (auto id : deadIdList) {
-            auto tmpId        = id - decrease;
-            Types::Dead &dead = deadList[tmpId];
+        Registry &registry = Registry::getInstance();
+        auto deadList      = registry.getComponents<Types::Dead>();
+        auto deadIdList    = deadList.getExistingsId();
+        Clock &clock       = registry.getClock();
+        auto it            = deadIdList.begin();
+
+        while (it != deadIdList.end()) {
```
Comment (guillaumeAbel) [inline src/ECS/Systems/Systems.cpp]: ta pas testé avec cette list d'id triéé ??

Comment (guillaumeAbel) [reply src/ECS/Systems/Systems.cpp]: et dcp pk supprimé le decrease ??

File: src/ECS/Systems/Systems.cpp [hunk 3]
```diff
@@ -346,14 +346,15 @@ namespace Systems {
...
         Registry::components<Types::Dead> arrDead = Registry::getInstance().getComponents<Types::Dead>();
-        std::size_t decrease                      = 0;
 
         std::vector<std::size_t> ids = arrHealth.getExistingsId();
-        for (auto &id : ids) {
-            auto tmpId = id - decrease;
-            if (arrHealth.exist(tmpId) && arrHealth[tmpId].hp <= 0) {
-                sendEnemyDeath(tmpId);
-                executeDeathFunction(tmpId, arrDead, decrease);
+        for (auto it = ids.begin(); it != ids.end();) {
```
Comment (guillaumeAbel) [inline src/ECS/Systems/Systems.cpp]: pas triée + pas decrease

**Review by guillaumeAbel — CHANGES_REQUESTED** [review]

**Review by guillaumeAbel — APPROVED** [review]

**Pull request merged**